volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+
+from struct import unpack
+from typing import Dict
+
+from volatility3.framework import constants, renderers, interfaces
+from volatility3.framework.configuration import requirements
+from volatility3.framework.exceptions import PagedInvalidAddressException
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import conversion, format_hints
+from volatility3.plugins import yarascan
+
+vollog = logging.getLogger(__name__)
+
+try:
```

Comment:
Since we import yarascan, this check for yara *should* be done for us.  I'm ok having this code duplicated if we also need to pull in the yara module, but it appears we only use it for yara.compile, which seems to be something exposed by the yarascan plugin in `process_yara_options`.  If that method isn't suitable, I'd prefer to extend it/fix it rather than pulling in the compilation to this plugin and requiring each plugin that uses yara to duplicate this check...

Reply:
The way the `process_yara_options` is written right now you can pass it one of 3 things:

- A single String or single regex
- Path to a yara text file
- path to a yara compiled binary file. 

It's not possible to pass it a rule as a string. 

We could add another option to yarascan in order to accept a full rule as an argument and then update this to match? 

Happy to make that change and PR it.

Reply:
Alternative woudl be to write the yara rule for MFT records as a regex, but this will slow down scanning.

Reply:
The regexp route would reduce the dependency on yara, but if yara rules are a better way of doing it, then go that way.  I'm up for changing the process_yara_options method, we  just need to figure out if the change is additive (and only bumps the minor version number) or a significant change (in which case we should think through all the use cases and try to make sure it doesn't need change often again).  It somewhat feels as though the reading of the file could be done by the plugins, but I need to think about it a little to figure out the best API layout.  Always open for suggestions too...  5;)

Reply:
OK updated this to remove that dependency, still using yara but now uses the `process_yara_options`